provider/datagen/src/driver.rs

```diff
@@ -220,36 +476,67 @@ impl DatagenDriver {
             log::warn!("No keys selected");
         }
 
-        if matches!(fallback, FallbackMode::Preresolved) && locales.is_none() {
-            return Err(DataError::custom(
-                "FallbackMode::Preresolved requires an explicit locale set",
-            ));
+        // Apply settings from FallbackMode
+        match legacy_fallback_mode {
+            FallbackMode::PreferredForExporter => {
+                // No-op: use settings from locales_fallback
+            }
+            FallbackMode::Runtime => {
+                let config = locales_fallback.coerce_with_fallback();
+                config.deduplication_strategy = Some(DeduplicationStrategy::Maximal);
+                config.runtime_fallback_location = Some(RuntimeFallbackLocation::Internal);
+            }
+            FallbackMode::RuntimeManual => {
+                let config = locales_fallback.coerce_with_fallback();
+                config.deduplication_strategy = Some(DeduplicationStrategy::Maximal);
+                config.runtime_fallback_location = Some(RuntimeFallbackLocation::External);
+            }
+            FallbackMode::Preresolved => {
+                let config = locales_fallback.coerce_without_fallback();
```

Comment:
These coercions are pretty messy and intertwine legacy and non-legacy code paths. I think you should keep setting the `legacy_locales` field in the locale setters, and work with that if `legacy_fallback_mode` is used.

Reply:
done